---
layout: default
title: Research on Protein Mutations and Ligand Binding
---

<div class="main-content">
  <header>
    <h1>Research on Protein Mutations and Ligand Binding</h1>
    <p>This --test- investigates how mutations in protein sequences affect ligand binding affinities, focusing on reference-mutant groups with a shared ligand.</p>

  </header>

  <section id="introduction">
    <h2>Introduction</h2>
    <p> TO BE DONEEE</p>
  </section>

   <!-- Vertical Line-->
   <hr>

  <!-- Part 1: Mutant Analysis -->
  <section id="mutant-analysis">
    <h2>Mutants Analysis</h2>
    <p>
      Protein mutations play a pivotal role in understanding various biological processes, especially in drug development. These mutations can significantly influence how a protein interacts with specific molecules, known as ligands. Accounting for these variations is essential when designing drugs that aim to be effective for a broad population. This crucial interplay between mutations and ligand binding sparked our curiosity and inspired the focus of the mutant analysis. We extracted the five “interaction pairs” with the highest number of mutants from BindingDB to display and analyse. An “interaction pair” here refers to a ligand and a group of similar proteins, i.e. a wild-type (WT) protein and its mutants.

      It would be too long to analyse in details every single one of those five interactions, so we have decided to focus on one that we found interesting because it includes all kinds of mutations (insertions, deletions, and substitutions). Nevertheless, all plots are available for every interaction pairs and can be displayed by selecting the corresponding dropdown option. Here is a summary of the five interaction pairs.
    </p>
    <!-- Table of Interaction Pairs-->
    <table>
      <tr>
          <th>Pair n°</th>
          <th>Ligand Name</th>
          <th>WT Protein</th>
      </tr>
      <tr>
          <td>1</td>
          <td>N-[7-chloro-1-[(3R)-1-[(E)-4-(dimethylamino)but-2-enoyl]azepan-3-yl]benzimidazol-2-yl]-2-methylpyridine-4-carboxamide</td>
          <td>EGFR_HUMAN</td>
      </tr>
      <tr>
          <td>2</td>
          <td>(E)-N-[4-(3-chloro-4-fluoroanilino)-7-[(3S)-oxolan-3-yl]oxyquinazolin-6-yl]-4-(dimethylamino)but-2-enamide</td>
          <td>EGFR_HUMAN</td>
      </tr>
      <tr>
          <td>3</td>
          <td>N-(3-chloro-4-fluorophenyl)-7-methoxy-6-(3-morpholin-4-ylpropoxy)quinazolin-4-amine</td>
          <td>EGFR_HUMAN</td>
      </tr>
      <tr>
          <td>4</td>
          <td>4-[(1R)-1-aminoethyl]-N-pyridin-4-ylcyclohexane-1-carboxamide</td>
          <td>KAPCA_BOVIN</td>
      </tr>
      <tr>
          <td>5</td>
          <td>4-methyl-5-[[(2S)-2-methyl-1,4-diazepan-1-yl]sulfonyl]isoquinoline</td>
          <td>KAPCA_BOVIN</td>
      </tr>
    </table>
    <p>
      We briefly explain here the notation used to describe mutations. There are three types of mutations:
      <ul>
        <li> A <strong>substitution</strong> is a mutation that changes the type of amino acid at a given position. For example, T790M means that at position 790 on the target sequence, the amino acid Threonine was mutated to Methionine</li>
        <li> A <strong>deletion</strong> is a mutation that shortens the length of the protein. For example, [30-120] means that the protein was truncated and only the sequence from amino acid 30 to 120 remain. A deletion of a single amino acid at position 19 would be indicated by [1-18, 20-120]</li>
        <li> An <strong>insertion</strong> is a mutation that makes the protein longer by introducing amino acids. [1-769, ASV,770-1210] means that three amino acids were inserted between amino acid at position 769 and amino acid at position 770, respectively Alanine, Serine, and Valine</li>
      </ul>
    </p>
  </section>

  <section id="alignments">
    <h3>Multiple Sequence Alignment (MSA)</h3>
    <p>At first sight, identifying mutations from protein names might seem straightforward if you’re human. An example is EGFR[1-18, 20-1210, C797S], where there is a deletion at position 19 and a substitution at position 797 from Cysteine to Serine. However, inconsistent formatting, different brackets, notations, or errors (like EGFR[1-751, 75360-1210] instead of EGFR[1-751, 760-1210]) makes this task non-trivial for a computer.</p>

    <p>To solve this, we turned directly to the protein sequences. Using pairwise alignment with the Needleman-Wunsch algorithm, we compared each mutant sequence to the reference protein. This approach allowed us to identify insertions, deletions, and substitutions, overcoming the ambiguity hidden in the names.</p>

    <p>Once the differences were identified, we constructed a Multiple Sequence Alignment (MSA). The result is a clean and intuitive visualization: the reference sequence at the top, mutants neatly aligned below.</p>

    <p>To make the changes stand out, we added color:</p>
    <ul>
      <li><strong>Insertions</strong> are highlighted in <span style="color: rgb(21, 189, 18);">green</span>.</li>
      <li><strong>Substitutions</strong> are highlighted in <span style="color: rgb(224, 224, 18);">yellow</span>.</li>
      <li><strong>Deletions</strong> are highlighted in <span style="color: rgb(245, 61, 0);">red</span>.</li>
    </ul>
    

    <p>With this alignment, variations against the reference, and patterns across mutants, become instantly visible, offering insights into how these changes might relate to functional outcomes, like shifts in IC50 values. Pick an interaction pair, scroll through the protein sequences, and you’ll see the different types of mutations neatly aligned, ready to reveal their story.</p>

    <iframe
      src="public/assets/alignments.html"
      width="100%"
      height="270"
      style="border: none"
    ></iframe>
  </section>

  <section id="ic50-scatter">
    <h3>Impact of mutations on binding affinity</h3>
    <p>To visualize the impact of specific mutations on binding affinity, we decided to first use scatter plots for a given interaction pair Wild-Type protein - Ligand. To assess the impact on binding affinity we look at the log ratio of IC50 values between the WT and the mutant. IC50 is a measure that indicates how much of a given ligand is needed to inhibit, in vitro, a given biological process or biological component by 50%. This value was present in the majority of experiments included in BindingDB and this is why it was used as a measure of binding affinity in our project. Given the range of values in the dataset, using the log ratio was the best way to visualize the results that can be interpreted as such: a positive value means a decreased binding affinity and a negative value means an increased binding affinity.</p>
    <p>Many information can be found on the plot such as the position of the mutation on the amino acid sequence (x-axis) as well as complementary information like the name of the mutant and the type of mutation that can be found by going over certain points on the plot using the user’s computer mouse. This interactive plot allows the user to zoom in certain parts of the sequence, to keep only specific mutants, and many more features that we hope the user will discover and play with.</p>
    <p>The second plot below the one just described is very similar but is colored in a different way. Color is used in the first plot to identify all mutations that are in the same mutant, but it is used in the second plot to visualize the difference in ESM2 probability. ESM2 is a masked language model that can be used to predict the probability of having a specific amino acid at a given position on a target sequence. It is thus interesting to compare the “likelihood” of the mutated amino acid versus the WT amino acid for a given substitution. This process can not be done for deletions and insertions since ESM2 is trained specifically to predict amino acids at masked positions. Since probabilities are between 0 and 1 by definition, the difference of the mutated one minus the WT one is bound between -1 and 1. A difference of -1 indicates that ESM2 is absolutely sure that the WT amino acid needs to be at the given position for the protein and correspondingly, a difference of +1 indicates that ESM2 is very confident about placing the mutated amino acid at the given position. The two scatter plots are after another to allow a better transition between both during analysis. As for the first plot, the user can use the computer mouse to go over points on the plot to learn more about the mutation.</p>
    
    <iframe src="public/assets/interactive_ic50_plot.html" width="120%" height="720" style="border:none;"></iframe>
    
    <p>Several interesting insights are observable on the subplots for the <strong>Interaction pair 2</strong> (EGFR_HUMAN). We see that the two first mutants (<span style="color: #4361ee;">[1-18, 20-1210, C797S]</span>, <span style="color: #f75c03;">[1-18, 20-1210, T790M, C797S]</span>) are very similar in terms of mutations, i.e., the second mutant has one extra point mutation at position 790 (from T to M). Nevertheless, we see a big difference on the y-axis between those two mutants. We can thus understand that the mutation at position 790 results in a great reduction of the binding affinity (even if most of the mutations on this plot are non-beneficial). The same exact conclusion can be drawn by looking at mutants 3 and 4 (<span style="color: #f1c40f;">[1-18, 20-1210, T790M]</span>, <span style="color: #ff0054;">[1-18, 20-1210]</span>), mutants 8 and 9 (<span style="color: #720026;">[L858R, C797S]</span>, <span style="color: #808000;">[L858R, T790M, C797S]</span>), and mutants 10 and 11 (<span style="color: #4361ee;">[L858R]</span>, <span style="color: #4cc9f0;">[T790M, L858R]</span>) Moreover, we also notice that for the mutant with the long deletion (<span style="color: #1d7874;">[668-1210,T790M,L858R]</span>), the IC50 log ratio is exafctly the same as for the mutant <span style="color: #4cc9f0;">[T790M, L858R]</span>. This indicates that the deletion has no direct impact on the binding affinity..</p>
    <p> A next step for this project could be to dig further in this given interaction and visualize the 3D structure of the binding to understand the importance of the mentioned position. This could be used to validate experimentally found results. Also, the only two beneficial mutations are the mutants <span style="color: #ff0054;">[1-18, 20-1210]</span> (deletion at position 19) and <span style="color: #4361ee;">[L858R]</span> (Leucine to Arginine at position 858).</p>
    <p> The bottom plot displays three mutations:</p>
      <ul>
        <li>mutation at position 790 from Threonine to Methionine: the probability difference is around -0.2 meaning Threonine is more probable than Methionine according to ESM2. The difference being small, it is hard to draw any conclusions but the negative sign still indicates a similar idea as our previous plot in which we said that T790M was non-beneficial</li>
        <li>mutation at position 797 from Cysteine to Serine: the probability difference is around +0.3 indicating that Serine is more likely to be at that position compared to Cysteine according to ESM2</li>
        <li>mutation at position 858 from Leucine to Arginine: the probability difference is around -0.8 which strongly indicates that Leucine is more likely to be at that position compared to Arginine. This is partially in contradiction with our previous analysis that stated that this mutation was beneficial</li>
      </ul>
    <p> Overall, as we will see in our other analysis, binding affinity is not properly captured by ESM2 and these results should be taken with a grain of salt.</p>
  </section>

  <section id="barplot-IC50">
    <h3>Impact of ligands on binding affinity</h3>
    <p>So far, we looked at a given interaction pair Ligand - WT protein and explored how mutations in the WT protein affect binding affinity. Interestingly, among the five pairs we selected, there are only two WT proteins: EGFR_HUMAN and KAPCA_BOVIN. This opens up an intriguing opportunity to flip the perspective and examine how different ligands impact binding affinity for a given pair WT protein - Mutant. We collected data for the five interaction pairs, paving the way for a fresh angle of analysis. </p>
    <p>For better visualization, we produced two bar plots—one for each WT protein—showcasing how different ligands influence binding affinity across their mutant variants. The bar plots show again the log ratio of IC50 values and each group of bars represents this value for a given WT/mutant pair. For a WT-Mutant pair, the colors of the bars identify the interacting ligand </p>

    <div class="plots-container">
      <!-- First Plot -->
      <div class="plot-item">
        <img src="public/assets/EGFR_HUMAN/barplot_EGFR_HUMAN.png" alt="Barplot EGFR">
      </div>
      <!-- Second Plot -->
      <div class="plot-item">
        <img src="public/assets/KAPCA_BOVIN/barplot_KAPCA_BOVIN.png" alt="Barplot KAPCA">
      </div>
    </div>
    <p>The goal was to see if any clear trends arose from the plot. Logically, this is not the case, since binding affinity depends on many components and not only the properties of the ligand.</p>
    <p>Still, for the two last mutants shown on the left plot we observe two very different patterns even though there is only one different mutation. We can thus make multiple hypothesis such as:</p>
        <ul>
          <li> position 790 is not important or not involved in the binding of the protein EGFR (and its mutants) with ligand 1</li>
          <li> the mutation T790M reduces the binding affinity with ligand 2 and ligand 3 </li>
        </ul>
    <p>Similarly, for the two last mutants shown on the right plot we see a big difference on the log ratio even though there is only one different mutation. We can thus make multiple hypothesis such as:</p>
        <ul>    
          <li> position 183 is important or involved in the binding of the protein KAPCA (and its mutants) with both ligand 4 and 5</li>
          <li> the mutation T183A reduces the binding affinity with ligand 4 and 5</li>
        </ul>
  </section>

  <!-- 3D Protein Structures -->
  <section id="side-mutants-plot">
    <h3>3D visualizations of proteins</h3>
    <p>

      Enough of long protein names and and numbers—we’ve talked plenty about how changes in protein structure affect binding affinity. Now, it’s time to see the proteins themselves in action! The next interactive plots let you explore the actual 3D structures of two proteins and some of their mutants. For the mutants, <span style="color: #636363;">grey</span> parts represent deletions, and <span style="color: #008000;">green</span> spheres represent substitutions. Interestingly, we see that all the mutations are located in a similar subspace for the mutant KAPCA [L49I, V123M, E127D, Q181K, T183A], and a further analysis could look at the importance of this part of the structure for binding affinity.
     </p>
    <div class="mutants-container">
      <!-- First Mutant Dropdown and Plot -->
      <div class="EGFR-mutant-item">
        {% include_relative public/assets/EGFR_HUMAN/EGFR_mutants.html %}
      </div>
      <!-- Second Mutant Dropdown and Plot -->
      <div class="KAPCA-mutant-item">
        {% include_relative public/assets/KAPCA_BOVIN/KAPCA_mutants.html %}
      </div>
    </div>
  </section>

  <!-- Visualization of Ligands' 3D Structure -->
  <section id="3d-plot-ligand">
    <h3>3D visualizations of ligands</h3>
    <p>
      For completeness, we have decided to also include the 3D visualization of the ligands. We hinted at the importance of the ligand in binding affinity even though we mostly focused on the importance of the protein, and another interesting project (maybe for next year ADA students) would be to investigate the structure of ligands in more details.
    </p>
    {% include_relative public/assets/ligand_representation_uni-box.html %}
  </section>

  <!-- Conclusion mutant analysis -->
  <section id="conclusion-mutants">
    <h3>Final words on Mutant Analysis</h3>
    <p> We have seen that we were able to identify key mutations sites for binding affinity and find examples of interactions where the ligand had an important impact on binding affinity. A next project could be to analyse the properties of the ligands and the mutated amino acids to try to find possible explanations for this impact. As mentioned earlier, looking at the 3D structure of the interaction could also validate some of the results we have found with our analysis.</p>
  </section>
  <!-- Vertical Line-->
  <hr>

  <!-- Part 2: Embeddings Analysis-->

  <section id="embeddings-analysis">
    <h2>Embeddings Analysis</h2>
    <p>
      Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code.
    </p>
    <iframe src="public/assets/embeddings/3d_combined_clustering_plot.html" width="100%" height="500" style="border:none;"></iframe>
  </section>

  <!-- Bipartite Graph-->
  <section id="embeddings-bipartite-graph">
    <h3>Abi's bipartite graph</h2>
    <p>
      Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code.
    </p>
    <img 
      src="public/assets/embeddings/bipartite_graph.png" 
      alt="Bipartite Graph" 
      style="width:100%; height:auto; display:block; margin: 0 auto;" />
  </section>

  <!-- Unipartite Graph-->
  <section id="embedding-unipartite-graphs">
    <h3>Abi's unipartite graphs</h3>
    <p>
      Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code.
    </p>
    <div class="plots-container">
      <!-- First Plot -->
      <div class="plot-item">
        <img src="public/assets/embeddings/unipartite_graph_ligands.png" alt="Ligand Unipartite Graph">
      </div>
      <!-- Second Plot -->
      <div class="plot-item">
        <img src="public/assets/embeddings/unipartite_graph_proteins.png" alt="Protein Unipartite Graph">
      </div>
    </div>
  </section>  

  <!-- Regression Plots-->
  <section id="embedding-regression-plots">
    <h3>Abi's regression plots</h3>
    <p>
      Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code. Fill in your text here and organize the index.html code.
    </p>
    <div class="plots-container">
      <!-- First Plot -->
      <div class="plot-item">
        <img src="public/assets/embeddings/regression_plot_ligand.png" alt="Ligand Regression Plot">
      </div>
      <!-- Second Plot -->
      <div class="plot-item">
        <img src="public/assets/embeddings/regression_plot_protein.png" alt="Protein Regression Plot">
      </div>
    </div>
  </section>

 <!-- Vertical Line-->
 <hr>

 <!-- Conclusion-->
  <section id="conclusion">
    <h2>Conclusion</h2>
    <p>
      To be AADEDDEDDEDED
    </p>
  </section>
</div>
